### ISSUE TITLE
Add ability to "backup" to network, and restore from said backup (USMT-like functionality)

### ISSUE THREAD

**Issue by staze** [IBM/mac-ibm-migration-tool#9]
**Is your feature request related to a problem? Please describe.**
We often have situations where a tech needs to wipe and restore a machine, or retain a backup of an old machine. 

**Describe the solution you'd like**
It would be awesome if this tool would allow backing up to a DMG on a network share and then restoring from one of those backups. 

**Describe alternatives you've considered**
Time machine backups, which are time consuming to set exclusions, etc

**Additional context**
Basically just looking for a Mac equivalent of Microsoft's USMT: https://learn.microsoft.com/en-us/windows/deployment/usmt/usmt-overview
